src/meta/meta_ckpt.c

```diff
@@ -1262,6 +1262,198 @@ __wt_meta_ckptlist_update_config(
     return (ret);
 }
 
+/*
+ * __ckpt_extract_blkmod_bitmap --
+ *     Extracts the blkmod info in a checkpoint into a bitmap within a WT_ITEM.
+ */
+static int
+__ckpt_extract_blkmod_bitmap(
+  WT_SESSION_IMPL *session, WT_CKPT *ckpt, const char *id, WT_ITEM *output_bitmap)
+{
+    WT_BLOCK_MODS *blk;
+    u_int i;
+
+    WT_CLEAR(*output_bitmap);
+
+    for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk)
+        if (F_ISSET(blk, WT_BLOCK_MODS_VALID) && strcmp(id, blk->id_str) == 0)
+            WT_RET(__wt_buf_set(session, output_bitmap, blk->bitstring.data, blk->bitstring.size));
+
+    return (0);
+}
+
+/*
+ * __get_blkmods --
+ *     Extracts the blkmod info from the metadata about a file+id into a bitmap within a WT_ITEM.
+ */
+static WT_INLINE int
+__get_blkmods(WT_SESSION_IMPL *session, const char *uri, const char *id, WT_ITEM *output_item)
+{
+    WT_CONFIG blkconf;
+    WT_CONFIG_ITEM backup_config_value, blocks, blocks_key, blocks_value, key;
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    char *file_config;
+
+    WT_CLEAR(backup_config_value);
+    WT_CLEAR(blocks);
+    WT_CLEAR(blocks_key);
+    WT_CLEAR(blocks_value);
+    WT_CLEAR(key);
+    WT_CLEAR(*output_item);
+
+    metadata_cursor = NULL;
+    file_config = NULL;
+
+    WT_RET(__wt_metadata_cursor_open(session, NULL, &metadata_cursor));
+
+    metadata_cursor->set_key(metadata_cursor, uri);
+    WT_ERR(metadata_cursor->search(metadata_cursor));
+    WT_ERR(metadata_cursor->get_value(metadata_cursor, &file_config));
+    WT_ERR(
```

Comment:
Can we get `WT_NOTFOUND` here? Is it possible or would that be an issue?
If it's possible, the `WT_ERR` would be incorrect I believe.

Reply:
Hi @etienneptl , that's a great question, and I think it's at least theoretically possible. The solution is to move the 
`    if (ret == WT_NOTFOUND)
        ret = 0;`
after the 'err' label.

Reply:
I am afraid this could catch unwanted errors if this function ever changes.
I would prefer:
```
if (ret == WT_NOTFOUND) {
  /* Nothing to check. we are done. */
  ret = 0;
  goto err;
}
```

Reply:
Since 3952c6e3cbccedccc8f6ba3f3303528ebef1d4fa, you are now hiding a potential error from the call `WT_ERR(metadata_cursor->search(metadata_cursor));`.

Reply:
@sueloverso, what are your thoughts on this?

Reply:
@etienneptl and @sueloverso , thanks for the feedback. I've moved the check for `WT_NOTFOUND` back to before the `err` label, and added an assert for WT_NOTFOUND on the result of `metadata_cursor->search` as suggested.